pkg/archive/archive.go

```diff
@@ -487,11 +487,14 @@ func DownloadAndExtractTarball(url string, removeTopLevel bool) (string, func(),
 		_ = os.Remove(tarball)
 	}()
 
-	err = util.DownloadFile(tarball, url, true)
+	err = util.DownloadFile(tarball, url, true, "")
 	if err != nil {
 		return "", nil, fmt.Errorf("unable to download %v: %v", url, err)
 	}
 	extractedDir, cleanup, err := ExtractTarballWithCleanup(tarball, removeTopLevel)
+	if err != nil {
+		_ = os.RemoveAll(extractedDir)
```

Comment:
I worry that this could be questionable if the extractedDir were assumed to already exist and have files in it.